services/types.go

```diff
@@ -454,3 +459,77 @@ type HeartbeatService interface {
 	// Watch watches the heartbeats for a service
 	Watch() (xwatch.Watch, error)
 }
+
+// ElectionOptions configure specific election-scoped options.
+type ElectionOptions interface {
+	// Duration after which a call to Leader() will timeout if no response
+	// returned from etcd. Defaults to 30 seconds.
+	LeaderTimeout() time.Duration
+	SetLeaderTimeout(t time.Duration) ElectionOptions
+
+	// Duration after which a call to Resign() will timeout if no response
+	// returned from etcd. Defaults to 30 seconds.
+	ResignTimeout() time.Duration
+	SetResignTimeout(t time.Duration) ElectionOptions
+
+	// DefaultValue returns the value that will be placed in the key of the
+	// leader of the election iff (1) no CampaionOptions option is passed to
+	// Campaign() and (2) the call to os.Hostname() fails. This value will very
+	// rarely be used since os.Hostname() should be reliable, and defaults to
+	// "default_hostname".
+	DefaultValue() string
+	SetDefaultValue(s string) ElectionOptions
+
+	// Hostname returns the hostname of the host if accessible, otherwise the
+	// value for DefaultValue().
+	Hostname() string
+
+	String() string
```

Comment:
As discussed above probably makes sense to rename it to `ID` to be more explicit about its purpose.